Tasks/UniversalPackagesV0/universalpublish.ts

```diff
@@ -153,6 +153,8 @@ function publishPackageUsingArtifactTool(publishDir: string, options: artifactTo
     const execResult: IExecSyncResult = artifactToolRunner.runArtifactTool(options.artifactToolPath, command, execOptions);
 
     if (execResult.code === 0) {
+        console.log(`##vso[task.setvariable variable=UniversalPackage_Name;]${options.packageName}`);
```

Comment:
I've always had a concern about the singleton nature of the task, and wanted to get a 'lock file' type of manifest supported too.  Under multi-run or multi-package, a single variable starts to lose value.  I know this is publish not download, but am still thinking a bigger "what I've done" parseable/mergable summary would be nice, but could be overkill for right now.  I don't want to get locked into permanent expectation of a singleton variable without a bit more thought though.  + @zarenner / @mitchdenny.

Reply:
Agree - that said, there's nothing really preventing the envvar (or multiple envvars) from containing a list of packages published. Perhaps differently named. I'll let Mitch comment since I'm sure he has relevant thoughts / future plans.

Reply:
To be honest, I think that these more advanced multi-upload scenarios may end up being done with scripts. You can only get so far with a set-orientated upload of multiple packages before you start needing to think about branching logic etc which all screens scripting.

At which point Universal Packages emitting JSON like the AZ CLI seems appropriate (especially if we support ```--query``` syntax.

Reply:
I've updated the code to take an input variable as part of the task to store the outputted published packaged name.  I think if we want to revisit this in the future to use a json file that seem reasonable.